app/assets/javascripts/service_types/IsServiceWorking.js

```diff
@@ -0,0 +1,186 @@
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      serviceTypeSelected: null // options: null, 502, 503
+    };
+
+    this.fetchIsServiceWorkingData = this.fetchIsServiceWorkingData.bind(this);
+    this.renderPage = this.renderPage.bind(this);
+  }
+
+  fetchIsServiceWorkingData() {
+    const {serviceTypeSelected} = this.state;
+
+    const url = `/api/is_service_working_json/${serviceTypeSelected.id}`;
+
+    return apiFetchJson(url);
+  }
+
+  render() {
+    return (
+      <div className="IsServiceWorking">
+        <ExperimentalBanner />
+        <SectionHeading>
+          "Is this Service Working for this Student?"
+        </SectionHeading>
+        <p style={{padding: '10px 0 0 10px'}}>
+          Every student is different.
+          One student might respond well to a service, while another might respond poorly.
+        </p>
+        <p style={{padding: '10px 0 10px 10px'}}>
+          Use the buttons below to explore how students may have responded to
+          services assigned to them over the past school year.
+        </p>
+        {this.renderButtons()}
+        {this.renderCharts()}
+      </div>
+    );
+  }
+
+  renderButtons() {
+    const attendanceOfficer = {
+      id: 502,
+      name: 'Attendance Officer',
+    };
+
+    const attendanceContract = {
+      id: 503,
+      name: 'Attendance Contract',
+    };
+
+    return (
+      <div style={style.buttonSectionStyle}>
+        <Button style={style.buttonStyle}
+          onClick={() => {
+            this.setState({serviceTypeSelected: null}, () => {
+              this.setState({serviceTypeSelected: attendanceOfficer});
+            });
+          }}>
+          Attendance Officer
+        </Button>
+        <Button style={style.buttonStyle}
+          onClick={() => {
+            this.setState({serviceTypeSelected: null}, () => {
+              this.setState({serviceTypeSelected: attendanceContract});
+            });
+          }}>
+          Attendance Contract
+        </Button>
+      </div>
+    );
+  }
+
+  renderCharts() {
+    const {serviceTypeSelected} = this.state;
+
+    if (!serviceTypeSelected) return null;
+
+    return (
+      <GenericLoader
+        promiseFn={this.fetchIsServiceWorkingData}
+        render={this.renderPage} />
+    );
+  }
+
+  renderPage(json) {
+    return (
+      <div style={style.container}>
+        {this.renderStudents(json)}
+      </div>
+    );
+  }
+
+  renderPhasebands(services) {
+    return services.map(service => {
+      return {
+        momentUTCstart: moment.utc(service.date_started),
+        momentUTCend: moment.utc(service.discontinued_at),
+        text: this.state.serviceTypeSelected.name
+      };
+    }, this);
+  }
+
+  renderStudents(json) {
+    const chartData = json.chart_data;
+    const sortedData = _.sortBy(chartData, datum => {
+      return (datum.services.length > 0)
+        ? _.map(datum.services, 'date_started').sort()[0]
```

Comment:
Will this sort correctly?  I'm assuming `date_started` is a string and that this is sorting by string order.

If it is string order, `datum.services.map(service => toMomentFromTime(service.date_started).getDate()).sort()[0]` might do the trick to convert the Rails timestamp to a JS `Date`.